plugins/modules/ndo_tenant_mld_snooping_policy.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Akini Ross (@akinross) <akinross@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_tenant_mld_snooping_policy
+short_description: Manage Multicast Listener Discovery (MLD) Snooping Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage MLD Snooping Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Akini Ross (@akinross)
+options:
+  template:
+    description:
+    - The name of the tenant template.
+    type: str
+    aliases: [ tenant_template ]
+    required: true
+  name:
+    description:
+    - The name of the MLD Snooping Policy.
+    type: str
+    aliases: [ mld_snooping_policy ]
+  uuid:
+    description:
+    - The UUID of the MLD Snooping Policy.
+    - This parameter is required when the MLD Snooping Policy O(name) needs to be updated.
+    type: str
+  description:
+    description:
+    - The description of the MLD Snooping Policy.
+    - Providing an empty string will remove the O(description="") from the MLD Snooping Policy.
+    type: str
+  admin_state:
+    description:
+    - The administrative state of the MLD Snooping Policy.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  fast_leave_control:
+    description:
+    - The fast leave control of the MLD Snooping Policy.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  querier_control:
+    description:
+    - The querier control of the MLD Snooping Policy.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  querier_version:
+    description:
+    - The querier version of the MLD Snooping Policy.
+    - Defaults to C(v2) when unset during creation.
+    type: str
+    choices: [ v1, v2 ]
+  query_interval:
```

Comment:
Should we also include the time period of all the interval options?
1. query_interval
2. query_response_interval
3. last_member_query_interval
4. start_query_interval

Reply:
will add

@anvitha-jain nothing is mentioned in UI for this, refer to https://www.cisco.com/c/en/us/td/docs/dcn/ndo/4x/articles-441/nexus-dashboard-orchestrator-aci-tenants-and-tenant-policies-templates-441.html for the metrics